src/build.c

```diff
@@ -879,23 +879,17 @@ static GPid build_spawn_cmd(GeanyDocument *doc, const gchar *cmd, const gchar *d
 }
 
 
-/* Returns: NULL if there was an error, or the working directory the script was created in.
- * vte_cmd_nonscript is the location of a string which is filled with the command to be used
- * when vc->skip_run_script is set, otherwise it will be set to NULL */
-static gchar *prepare_run_script(GeanyDocument *doc, gchar **vte_cmd_nonscript, guint cmdindex)
+/* Returns: NULL if there was an error, or the working directory from which the script is executed.
+ * run_cmd is the location of a string which is filled with the command to be executed. */
+static gchar *prepare_run_script(GeanyDocument *doc, gchar **run_cmd, guint cmdindex)
```

Comment:
it probably should be mentioned that `run_cmd` will have different encoding depending on some arbitrary values (`vc->skip_run_script`).  Or even make it consistent, even though it'd require converting back and forth which is suboptimal.
Anyway, as it's only in utf-8 when it's a command and not a script it might be OK to just mention this.